core/components/molecules/navigation/navigation-item.js

```diff
@@ -48,7 +48,7 @@ const processIcon = (children, selected, inSubMenu) =>
 class NavigationItem extends React.Component {
   constructor(props) {
     super(props)
-    this.state = { subMenuOpen: props.defaultOpen || false }
+    this.state = { subMenuOpen: props.defaultOpen || null }
```

Comment:
why don't we default the initial state value from the `selected` props?
we could create a method getter...

```js
get isSelected() {
  const { selected, subMenu } = this.props;
  return selected || findSelectedSubItem(subMenu);
}

// and then just call
this.state = { subMenuOpen: defaultOpen || this.isSelected };
// ...
```